### ISSUE TITLE
Line2D markfacecolor=None fails if set_alpha() is called

### ISSUE THREAD

**Issue by TD22057** [matplotlib/matplotlib#11113]
### Bug report

Calling Line2D with markerfacecolor='None' should yield a marker with just an outline.  That works fine.  But if a call to artist.set_alpha( X ) where is X is anything but None is made, then the marker is filled no matter what with black color and alpha=X.

**Code for reproduction**

```python
import pylab as p
p.ion()
a=p.plot( [1,2], marker='o' )[0]

# Works, marker is unfilled
a.set_markerfacecolor( 'None' )

# Fails - marker is filled with grey
a.set_alpha( 0.5 )
```

**Actual outcome**

![alpha_bug](https://user-images.githubusercontent.com/1421685/39155540-47de3c06-4707-11e8-8f36-9450ec99ad51.png)

**Matplotlib version**
  * Operating system: RedHat Linux 7
  * Matplotlib version: 2.2.2
  * Matplotlib backend (`print(matplotlib.get_backend())`):  Qt5Agg
  * Python version: 2.7

**Comment by ImportanceOfBeingErnest** [comment]
Should be the same as #11104

**Comment by jklymak** [comment]
Yeah, I'll close as a duplicate, but feel free to comment on #11104 and the fix in #11105
